src/plugins/data_source_management/public/components/data_source_selectable/data_source_selectable.tsx

```diff
@@ -81,67 +81,101 @@ export class DataSourceSelectable extends React.Component<
     this.setState({ ...this.state, isPopoverOpen: false });
   }
 
+  handleSelectedOption(dataSourceOptions: DataSourceOption[], defaultDataSource: string | null) {
+    const [{ id }] = this.props.selectedOption!;
+    switch (id) {
+      case '':
+        this.setState({
+          ...this.state,
+          dataSourceOptions,
+          selectedOption: [{ id: LocalCluster.id, label: LocalCluster.label, checked: 'on' }],
+          defaultDataSource,
+        });
+        this.props.onSelectedDataSources([LocalCluster]);
+        break;
+      case undefined:
+        this.props.notifications.addWarning(
+          i18n.translate('dataSource.fetchDataSourceError', {
+            defaultMessage: 'Data source id is undefined',
+          })
+        );
+        break;
+      default:
+        if (!dataSourceOptions.find((ds) => ds.id === id)) {
+          this.props.notifications.addWarning(
+            i18n.translate('dataSource.fetchDataSourceError', {
+              defaultMessage: 'Data source with id is not available',
+            })
+          );
+          return;
+        }
+        break;
+    }
+  }
+
+  handleDefaultDataSource(dataSourceOptions: DataSourceOption[], defaultDataSource: string | null) {
+    const selectedDataSource = getDefaultDataSource(
+      dataSourceOptions,
+      LocalCluster,
+      defaultDataSource,
+      this.props.hideLocalCluster
+    );
+
+    if (selectedDataSource.length === 0) {
+      this.props.notifications.addWarning('No connected data source available.');
```

Comment:
When selectedDataSource.length === 0, should we also use this callback function this.props.onSelectedDataSources([])? then plugins team know it is [] and show corresponding page?